library/rsa.c

```diff
@@ -286,28 +272,26 @@ int mbedtls_rsa_deduce_private( mbedtls_mpi *P, mbedtls_mpi *Q,
     }
 
     mbedtls_mpi_init( &K );
+    mbedtls_mpi_init( &L );
```

Comment:
For reference, you don't need `L`.

```
diff --git a/library/rsa.c b/library/rsa.c
index b0ba1eb..a9ed796 100644
--- a/library/rsa.c
+++ b/library/rsa.c
@@ -259,7 +259,7 @@ int mbedtls_rsa_deduce_private( mbedtls_mpi const *P,
                                 mbedtls_mpi *D )
 {
     int ret = 0;
-    mbedtls_mpi K, L;
+    mbedtls_mpi K;
 
     if( D == NULL || mbedtls_mpi_cmp_int( D, 0 ) != 0 )
         return( MBEDTLS_ERR_MPI_BAD_INPUT_DATA );
@@ -272,17 +272,18 @@ int mbedtls_rsa_deduce_private( mbedtls_mpi const *P,
     }
 
     mbedtls_mpi_init( &K );
-    mbedtls_mpi_init( &L );
 
     /* Temporarily put K := P-1 and L := Q-1 */
     MBEDTLS_MPI_CHK( mbedtls_mpi_sub_int( &K, P, 1 ) );
-    MBEDTLS_MPI_CHK( mbedtls_mpi_sub_int( &L, Q, 1 ) );
+    MBEDTLS_MPI_CHK( mbedtls_mpi_sub_int( D, Q, 1 ) );
 
     /* Temporarily put D := gcd(P-1, Q-1) */
-    MBEDTLS_MPI_CHK( mbedtls_mpi_gcd( D, &K, &L ) );
+    MBEDTLS_MPI_CHK( mbedtls_mpi_gcd( D, &K, D ) );
 
     /* K := LCM(P-1, Q-1) */
-    MBEDTLS_MPI_CHK( mbedtls_mpi_mul_mpi( &K, &K, &L ) );
+    MBEDTLS_MPI_CHK( mbedtls_mpi_mul_mpi( &K, &K, Q ) ); //K := (P-1)*Q
+    MBEDTLS_MPI_CHK( mbedtls_mpi_sub_mpi( &K, &K, P ) ); //K := (P-1)*Q-P
+    MBEDTLS_MPI_CHK( mbedtls_mpi_add_int( &K, &K, 1 ) ); //K := (P-1)*(Q-1)
     MBEDTLS_MPI_CHK( mbedtls_mpi_div_mpi( &K, NULL, &K, D ) );
 
     /* Compute modular inverse of E in LCM(P-1, Q-1) */
@@ -291,7 +292,6 @@ int mbedtls_rsa_deduce_private( mbedtls_mpi const *P,
 cleanup:
 
     mbedtls_mpi_free( &K );
-    mbedtls_mpi_free( &L );
 
     return( ret );
 }
```

Given that bignum functions allocate under the hood in non-trivial ways, I don't know how much memory this saves, if any. And it makes the code more complex and very slightly slower. So I don't think it's worth changing.